pallets/capacity/src/tests.rs

```diff
@@ -1,9 +1,501 @@
+use super::*;
+use crate as pallet_capacity;
 use crate::mock::*;
-use frame_support::assert_ok;
+use frame_support::{assert_noop, assert_ok, traits::WithdrawReasons, BoundedVec};
+use testing_utils::{register_provider, run_to_block, staking_events};
 
 #[test]
-fn it_stakes() {
+fn stake_works() {
 	new_test_ext().execute_with(|| {
-		assert_ok!(Capacity::stake(RuntimeOrigin::signed(1), 1, 1));
+		let account = 200;
+		let target: MessageSourceId = 1;
+		let amount = 5;
+		register_provider(target, String::from("Foo"));
+		assert_ok!(Capacity::stake(RuntimeOrigin::signed(account), target, amount));
+
+		// Check that StakingAccountLedger is updated.
+		assert_eq!(Capacity::get_staking_account_for(account).unwrap().total, amount);
+		assert_eq!(Capacity::get_staking_account_for(account).unwrap().active, amount);
+		assert_eq!(Capacity::get_staking_account_for(account).unwrap().unlocking.len(), 0);
+
+		// Check that StakingTargetLedger is updated.
+		assert_eq!(Capacity::get_target_for(account, target).unwrap().amount, amount);
+		assert_eq!(Capacity::get_target_for(account, target).unwrap().capacity, amount);
+
+		// Check that CapacityLedger is updated.
+		assert_eq!(Capacity::get_capacity_for(target).unwrap().remaining, amount);
+		assert_eq!(Capacity::get_capacity_for(target).unwrap().total_available, amount);
+		assert_eq!(Capacity::get_capacity_for(target).unwrap().last_replenished_epoch, 1);
+
+		let events = staking_events();
+		assert_eq!(events.first().unwrap(), &Event::Staked { account, target, amount });
+
+		assert_eq!(Balances::locks(&account)[0].amount, amount);
+		assert_eq!(Balances::locks(&account)[0].reasons, WithdrawReasons::all().into());
+	});
+}
+
+#[test]
+fn stake_errors_invalid_target_when_target_is_not_registered_provider() {
+	new_test_ext().execute_with(|| {
+		let account = 100;
+		let target: MessageSourceId = 1;
+		let amount = 1;
+		assert_noop!(
+			Capacity::stake(RuntimeOrigin::signed(account), target, amount),
+			Error::<Test>::InvalidTarget
+		);
+	});
+}
+
+#[test]
+fn stake_errors_insufficient_staking_amount_when_staking_below_minimum_staking_amount() {
+	new_test_ext().execute_with(|| {
+		let account = 000;
```

Comment:
What's the difference between this test and the one below? I see only a difference of MSA Id.

Reply:
Good catch; you are right. This is leftover from when we checked if the balance is less than zero.